10up/insecure-content-warning
# Post Locking

### Description of the Change

When insecure items are found, we stop publishing from happening unless the override checkbox is used. But the publish button is still active.

This PR utilizes the `lockPostSaving` function in the block editor and disables the publish button in the classic editor when insecure items are found. Both of these locks are removed if the override checkbox is checked or if all assets are fixed using the automated `Fix this` links.

### Alternate Designs

None

### Benefits

It's now more clear to the user that they can't publish the content unless they fix the assets or override the behavior. Also helps limit any potential saving that we don't want.

### Possible Drawbacks

If a user manually fixes the assets, there is no way to re-scan and remove the lock. In this scenario, they can utilize the override checkbox just fine but we may want to add a re-scan button.

### Verification Process

In either the block editor or classic editor:
1. Edit a post with insecure items in it
2. Try and publish the post
3. Notice the publish button is disabled
4. Check the override checkbox and notice the publish button is now enabled

### Checklist:


### Applicable Issues

Fixes #27
